### ISSUE TITLE
handleAgentCommand() method in DevdasCore?

### ISSUE THREAD

**Issue by B-T-Johnson** [wtepfenhart/Devdas#25]
The method handleAgentCommand(AgentMessage) from the type DevdasCore is never used; so what function does it provide?

DevdasCore has a handleSystemCommand(CommandMessage) method, which is used in the methods handleSystemBroadcast(CommandMessage) and processSystemCommand(CommandMessage). So is the handleAgentCommand(AgentMessage) method supposed to be used in a similar fashion?

- If that is the case, why does processAgentMessage(CommandMessage) ignore this method?
- Should there also be the parallel methods handleAgentBroadcast(AgentMessage) and handleAgentResponse(AgentMessage) to handle Agent messages that are broadcasted or are in response to other Agents?
- Are these abilities ignored for Agent processing because Agents should not have the same level of functionality as System commands?

**Comment by wtepfenhart** [comment]
Good catch! The handleAgentCommand was a method that I introduced initially along with the other methods you noted associated with the system. I was trying to do the same kind of thing for system and agent centric messages. All of the methods you suggested are missing were there. I realized that was kind of stupid on my part, so I ripped out most of it. The handleAgentCommand was used as a source to copy and paste code over to processAgentMessage and I went on my way without deleting it after the cut and paste job. 

Here's one way to think about the agents. They are interested in topics and direct communications that are usually in response to a request. I've handled the topics, but not the direct communications.
